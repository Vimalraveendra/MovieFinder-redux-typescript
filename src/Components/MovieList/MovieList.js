import React from "react";
import MovieCard from "../MovieCard/MovieCard";
import styles from "./MovieList.module.css";

// interface IMoviesListProps {}
// //here we are doing aggregation of all the props types.
// type propsList = IMoviesListProps & LinkStateProps;

const MovieList = ({ movies, error, searchField }) => {
  return error || searchField.length === 0 ? (
    <MovieCard error={error} />
  ) : (
    <div className={styles.movielist}>
      {movies &&
        movies.map(({ imdbID, Title, Poster, Year }) => {
          return Poster !== "N/A" ? (
            <MovieCard key={imdbID} Title={Title} Poster={Poster} Year={Year} />
          ) : null;
        })}
    </div>
  );
};

export default MovieList;
