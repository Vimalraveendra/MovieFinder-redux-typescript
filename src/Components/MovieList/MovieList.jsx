import React from "react";
import MovieCard from "../MovieCard/MovieCard";
import styles from "./MovieList.module.css";
import { connect } from "react-redux";

const MovieList = ({ movies, error }) => {
  console.log("movielist", error);
  console.log("movies", movies);
  return error && movies.length === 0 ? (
    <MovieCard error={error} />
  ) : (
    <div className={styles.movielist}>
      {movies &&
        movies.map(({ imdbID, Title, Poster, Year }) => {
          return (
            <MovieCard key={imdbID} title={Title} poster={Poster} year={Year} />
          );
        })}
    </div>
  );
};

const mapStateToProps = ({ moviesList: { movies, error } }) => ({
  movies,
  error,
});
export default connect(mapStateToProps)(MovieList);
