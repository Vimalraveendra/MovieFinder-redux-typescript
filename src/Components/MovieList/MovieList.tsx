import React from "react";
import MovieCard from "../MovieCard/MovieCard";
import styles from "./MovieList.module.css";
import { connect } from "react-redux";

import { createStructuredSelector } from "reselect";
import {
  selectMovies,
  selectError,
} from "../../Redux/FilmCard/FilmCard.selectors";
import { selectSearchField } from "../../Redux/SearchBox/SearchBox.selectors";
import { IMoviesList } from "../../Redux/FilmCard/FilmCard.types";
import { ISearchBox } from "../../Redux/SearchBox/SearchBox.types";

type propsList = IMoviesList & ISearchBox;

const MovieList = ({ movies, error, searchField }) => {
  return error || searchField.length === 0 ? (
    <MovieCard error={error} />
  ) : (
    <div className={styles.movielist}>
      {movies &&
        movies.map(({ imdbID, Title, Poster, Year }) => {
          return Poster !== "N/A" ? (
            <MovieCard key={imdbID} title={Title} poster={Poster} year={Year} />
          ) : null;
        })}
    </div>
  );
};

const mapStateToProps = createStructuredSelector({
  movies: selectMovies,
  error: selectError,
  searchField: selectSearchField,
});
export default connect(mapStateToProps)(MovieList);
