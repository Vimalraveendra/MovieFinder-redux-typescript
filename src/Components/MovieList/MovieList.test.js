import { shallow } from "enzyme";
import React from "react";
import MovieList from "./MovieList";

describe("MovieList component", () => {
  let wrapper, mockProps;
  // beforeEach it makes sure that it run before each one of the test
  beforeEach(() => {
    mockProps = {
      movies: [],
      searchField: "",
    };
    wrapper = shallow(<MovieList {...mockProps} />);
  });
  it("should render the MovieList Component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should return error if the searchField length is zero ", () => {
    expect(wrapper).toEqual({});
  });

  it("should return movies if the searchField length is greater than zero & movies exist", () => {
    const mockProps1 = {
      searchField: "Hello",
      movies: [
        {
          imdbID: "123",
          Title: "Hello",
          Poster: "how are you",
          Year: "2020",
        },
      ],
    };
    const wrapper1 = shallow(<MovieList {...mockProps1} />);
    expect(wrapper1.props().children.length).toEqual(1);
  });
});
