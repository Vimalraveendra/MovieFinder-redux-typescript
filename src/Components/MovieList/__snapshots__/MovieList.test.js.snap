// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList component should render the MovieList Component 1`] = `ShallowWrapper {}`;
